### PR TITLE
DEVX-2160: Added Self Balancing Clusters tutorial.

### PR DESCRIPTION
**Pull request by javabrett**
### Description 

https://confluentinc.atlassian.net/browse/DEVX-2160
Follows from https://confluentinc.atlassian.net/browse/DEVX-2162 and https://github.com/confluentinc/cp-demo/pull/353

_What behavior does this PR change, and why?_

Adds tutorial for executing the Self-Balancing Clusters demo.

Docs staging URL: http://staging-docs-independent.confluent.io.s3-website-us-west-2.amazonaws.com/docs-platform/PR/578/6.1.0/tutorials/cp-demo/docs/on-prem.html#self-balancing-clusters

### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

- [X] Documentation -->
- [X] Run cp-demo -->
- [ ] jmx-monitoring-stacks

### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._

### CODE REVIEW

**Review by Marcinthecloud — APPROVED** [review]
Looks great! Thanks Brett.

File: docs/on-prem.rst [hunk 1]
```diff
@@ -1330,6 +1330,64 @@ to setup alerts from there.
...
+#. Simulate a broker-failure by running the the script to kill the previously-added broker `kafka3`:
+
+   .. sourcecode:: bash
+
+      ./scripts/sbc/kill-broker.sh
+
+   This script will return when Self-Balancing Clusters has detected the broker failure and the recovery wait-time ``KAFKA_CONFLUENT_BALANCER_HEAL_BROKER_FAILURE_THRESHOLD_MS`` has expired, when self-healing has been triggered to reassign replicas on the ``kafka3`` broker to the remaining two brokers.
+   This has been set artificially-low for ``cp-demo`` to ``30000`` (30 seconds) in order to trigger self-healing promptly for the demo.
+
+#. Monitor the progress of replica-reassignment from the failed broker, which will eventually reduce the under-replicated partitions in the cluster back to zero.  To track completion of self-healing, run the following scripts:
```
Comment (ybyzek) [inline docs/on-prem.rst:1377]: Tell the user what they should expect in the output of these scripts.

File: docs/on-prem.rst [hunk 2]
```diff
@@ -1330,6 +1330,64 @@ to setup alerts from there.
...
+
+      ./scripts/sbc/add-broker.sh
+
+   The script will return when Self Balancing Clusters has acknowledged the broker-addition and started a rebalance task.
+
+#. Open Control Center at http://localhost:9021 and navigate to Brokers.  The Self-balancing panel will show 1 in-progress task.
+   Click on the panel and locate the in-progress add-broker task for broker with ID ``broker.3``.
+
+#. The add-broker rebalance task will progress through stages ``PLAN_COMPUTATION``, ``REASSIGNMENT`` and finally ``COMPLETED``.  The time spent in each phase will vary depending on the hardware you are running.
+   You can check the status using the following scripts to read the broker logs:
```
Comment (ybyzek) [inline docs/on-prem.rst:1351]: Tell the user what they should expect in the output of these scripts.

**Review by ybyzek — COMMENTED** [review]
Thanks for adding to the docs @javabrett .

**Review by arodoni — COMMENTED** [review]
Will approve when all @ybyzek 's comments are addressed.

File: docs/on-prem.rst [hunk 3]
```diff
@@ -1330,6 +1330,64 @@ to setup alerts from there.
...
+
+#. Run the following command to add a new broker ``kafka3`` to the cluster.
+
+   .. sourcecode:: bash
+
+      ./scripts/sbc/add-broker.sh
+
+   The script will return when Self Balancing Clusters has acknowledged the broker-addition and started a rebalance task.
+
+#. Open Control Center at http://localhost:9021 and navigate to Brokers.  The Self-balancing panel will show 1 in-progress task.
```
[the remark this answers is not in the document]
Comment (javabrett) [reply docs/on-prem.rst]: Added 3, may need review and editing.

File: docs/on-prem.rst [hunk 4]
```diff
@@ -1330,6 +1330,64 @@ to setup alerts from there.
 
 .. _cp-demo-monitoring:
 
+
+Self-Balancing Clusters
+-----------------------
+
+Confluent Platform Self-Balancing Clusters automates resource workload balancing across brokers in the cluster and provides self-healing in cases where a broker fails.  This simplifies scale-up and scale-down operations, ensuring workload is assigned to new brokers and automates recovery in case of a failure.
+
```
[the remark this answers is not in the document]
Comment (javabrett) [reply docs/on-prem.rst:1343]: Done.

**Review by ybyzek — CHANGES_REQUESTED** [review]
@javabrett I created a PR to merge into this PR --> https://github.com/confluentinc/cp-demo/pull/366 (lots of minor docs tweaks and a screenshot improvement).  I combed it over while I ran through the steps.  It would be great if you could please review and merge it in if you agree with the changes.

Also, in my environment, the `add-broker` was unsuccessful.  It failed to rebalance.  I think it's important to investigate and understand why -- functional issue or hardware resource issue?  If it failed in my environment (my laptop is relatively well spec'd), it will fail in others.

Screen Shot 2021-04-14 at 08 34 08 AM

**Comment by ybyzek** [comment]
@javabrett a second run-thru worked.  One difference between the two runs:

- run that failed: did not wait initial 15 minutes
- run that passed: did wait initial 15 minutes

I don't know if the initial 15 minute wait is the reason, but I did augment my PR to mention waiting

**Comment by ybyzek** [comment]
Cool, thanks for merging.  I triggered a new docs build (https://github.com/confluentinc/docs-platform/pull/578) -- let's confirm docs builds ok and then I'll approve this PR.

**Review by ybyzek — APPROVED** [review]
@javabrett the docs PR passes.  So this PR LGTM and ready to ship.  Wonderful work, thank you!

**Pull request merged**